utils/chsmack.c

```diff
@@ -125,12 +130,15 @@ static void modify_transmute(const char *path)
 	int rc;
 	switch (transmute_flag) {
 	case positive:
-		rc = follow_flag ?  stat(path, &st) : lstat(path, &st);
+		rc = follow_flag ? stat(path, &st) : lstat(path, &st);
 		if (rc < 0)
 			perror(path);
 		else if (!S_ISDIR(st.st_mode)) {
-			fprintf(stderr, "%s: transmute: not a directory\n",
-				path);
+			if (!recursive_flag) {
```

Comment:
Why not report an error on invocation like:
> chsmack --transmute --recureive some_file
?

You would warn on setting transmute for a file, but addition of --recursive flag changes the error to non-error?